File: library/src/main/java/mejust/frame/di/module/AppModule.java
package mejust.frame.di.module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import mejust.frame.net.HttpConfigHelper;

/**
 * @author : Beaven
 * @date : 2017-12-20 11:30
 * <p>
 * 全局Module，提供全局注入
 */
@Module
public class AppModule {

    private final Application application;

    public AppModule(Application application) {
        this.application = application;
    }

    /**
     * 提供应用全局的Application
     *
     * @return Application
     */
    @Singleton
    @Provides
    public Application provideApplication() {
        return application;
    }

    /**
     * 提供Http配置帮助类
     *
     * @return HttpConfigHelper
     */
    @Singleton
    @Provides
    public HttpConfigHelper provideHttpConfigHelper() {
        return new HttpConfigHelper();
    }
}

File: app/src/main/java/com/zhuazhu/frame/di/module/AppModule.kt
package com.zhuazhu.frame.di.module

import com.zhuazhu.frame.data.ApiOne
import dagger.Module
import dagger.Provides
import mejust.frame.di.component.FrameComponent
import javax.inject.Singleton

/**
 * @author wangpeifeng
 * @date 2018/06/04 15:32
 */
@Module
class AppModule {

    @Singleton
    @Provides
    fun provideApiOne(frameComponent: FrameComponent): ApiOne {
        return frameComponent.netManager().getApi(ApiOne::class.java)
    }

}